src/bp/ui-admin/src/Auth/index.ts

```diff
@@ -1,22 +1,46 @@
-import { ChatUserAuth } from 'common/typings'
+import { ChatUserAuth, StoredToken, TokenUser } from 'common/typings'
+import moment from 'moment'
 import ms from 'ms'
 
 import api from '../api'
 import history from '../history'
-
 export const TOKEN_KEY = 'bp/token'
 export const WORKSPACE_KEY = 'bp/workspace'
 export const CHAT_USER_AUTH_KEY = 'bp/chat_user_auth'
 const HOME_ROUTE = '/home'
 
-export function pullToken() {
-  const ls = localStorage.getItem(TOKEN_KEY)
-  return (ls && JSON.parse(ls)) || { token: null, expires: 0 }
+export const REFRESH_INTERVAL = ms('5m')
+export const MIN_MS_LEFT_BEFORE_REFRESH = ms('10m')
+
+export const getToken = (onlyToken: boolean = true): StoredToken | string | undefined => {
+  const token = localStorage.getItem(TOKEN_KEY)
```

Comment:
I'm not sure we should store auth tokens in localStorage it could be accessed by any scripts, right ?

Reply:
The token always was stored in local storage (or in cookie if storage is not available). I think we already had a discussion about that, but if we want to change it, I think it should be done outside of this PR.

I used the existing mechanism and did a small cleanup so the logic is identical on admin/studio

Reply:
You're right, let's keep this as is